v0/destinations/ga4/transform.js

```diff
@@ -606,14 +330,54 @@ function process(event) {
   const messageType = message.type.toLowerCase();
   let response;
   switch (messageType) {
+    case EventType.IDENTIFY:
+      response = [];
+      // 1. send login/signup event based on config
+      if (Config.sendLoginSignup) {
+        const traits = getFieldValueFromMessage(message, "traits");
+        const firstLogin = traits[`${Config.signupTrait}`];
+        if (!isDefinedAndNotNull(firstLogin)) {
+          throw new CustomError(
+            `[GA4] Idenitfy:: ${Config.signupTrait} is a required field in traits`,
+            400
+          );
+        }
+        if (firstLogin) {
+          message.event = "sign_up";
+        } else {
+          message.event = "login";
+        }
+
+        response.push(trackResponseBuilder(message, destination));
+      }
+
+      // 2. send generate_lead based on config
+      if (Config.generateLead) {
+        message.event = "generate_lead";
+        response.push(trackResponseBuilder(message, destination));
+      }
+
+      if (response.length === 0) {
+        throw new CustomError(
+          "[GA4] Idenitfy:: Server side identify is not enabled",
+          400
+        );
+      }
+
+      break;
     case EventType.TRACK:
       response = trackResponseBuilder(message, destination);
       break;
     case EventType.PAGE:
-      // passing page_view custom event for page()
+      // GA4 custom event (page_view) is fired for page
       message.event = "page_view";
       response = trackResponseBuilder(message, destination);
```

Comment:
Shouldn't we be pushing the output of `trackResponseBuilder` into `response` array?

Reply:
This doesn't apply for other message types apart from `identify` as `identify` can have `signup/login` or `generate_lead` event

Reply:
IMO keeping this way makes someone know that we are not trying to send multiple events here apart from identify.